apis/projects/v1alpha1/projecthook_types.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// A ProjectHookParameters defines the desired state of a Gitlab ProjectHook.
+type ProjectHookParameters struct {
+	URL *string `json:"url"`
+	// +optional
+	ConfidentialNoteEvents *bool `json:"confidential_note_events,omitempty"`
```

Comment:
We try to stick to Kubernetes API convention, so JSON field names should be camel case rather than snake case. Comment strings would be ideal too - we use them to generate user-facing documentation.

Reply:
thanks! I've updated the json fields and added comment strings